MapboxNavigation/CarPlayMapViewController.swift

```diff
@@ -102,35 +143,48 @@ class CarPlayMapViewController: UIViewController {
         resetCamera(animated: false, altitude: CarPlayMapViewController.defaultAltitude)
     }
     
-    override func viewWillDisappear(_ animated: Bool) {
+    override public func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
         styleObservation = nil
     }
     
-    public func zoomInButton() -> CPMapButton {
-        let zoomInButton = CPMapButton { [weak self] (button) in
-            guard let strongSelf = self else {
-                return
+    /**
+     Creates a new pan map button for the CarPlay map view controller.
+     
+     - parameter mapTemplate: The map template available to the pan map button for display.
+     */
+    @discardableResult public func createPanMapButton(for mapTemplate: CPMapTemplate) -> CPMapButton {
+        let panButton = CPMapButton { _ in
+            if !mapTemplate.isPanningInterfaceVisible {
+                mapTemplate.showPanningInterface(animated: true)
             }
-            strongSelf.mapView.setZoomLevel(strongSelf.mapView.zoomLevel + 1, animated: true)
         }
+        
         let bundle = Bundle.mapboxNavigation
-        zoomInButton.image = UIImage(named: "carplay_plus", in: bundle, compatibleWith: traitCollection)
-        return zoomInButton
+        panButton.image = UIImage(named: "carplay_pan", in: bundle, compatibleWith: traitCollection)
+        self.panMapButton = panButton
```

Comment:
An instance method shouldn’t create an object, store it in a property as a side effect, and return that object. The previous lazily computed property was more appropriate if we’re sure that it’s better to store the result, but if we go with a factory method, then it should have no side effects. (×2)